apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/Options.kt

```diff
@@ -267,3 +270,26 @@ class Options(
   }
 }
 
+/**
+ * Controls how scalar adapters are used in the generated code.
+ */
+@JsonClass(generateAdapter = true, generator = "sealed:type")
+sealed interface AdapterInitializer : Serializable
```

Comment:
Why does it need to be `Serializable` ?

Reply:
Got it. This is need by Gradle to snapshot the input properties. 

I always feel weird implementing `Serializable` since it's already serialized to Json. Also using Kotlin features like sealed interfaces and data classes in Gradle APIs feels dangerous too (think Groovy callers).

I would certainly have used 2 maps:

```kotlin
scalarsMapping: Map<String, String>
scalarAdapters: Map<String, String>
```

But I can't find a strong reason for the change so your choice!

Reply:
Yes sorry I should have commented :). I agree, I'm not very fond of making them as `Serializable` either. Using `String` feels safer, I'll make the change.